internal/ceb/virtual.go

```diff
@@ -140,50 +144,49 @@ func (v *Virtual) RunExec(ctx context.Context, h VirtualExecHandler, count int)
 			return err
 		}
 
-		// The idea here is that we're going to gather up all the configs sent down
-		// and then use them in an exec session when it's requested.
+		// Always update our config vars even if we have no exec so that
+		// while we're waiting for exec we're at least gathering config.
+		w.UpdateSources(ctx, msg.Config.ConfigSources)
+		w.UpdateVars(ctx, msg.Config.EnvVars)
 
-		for _, src := range msg.Config.ConfigSources {
-			dynamicSources[src.Type] = src
+		if msg.Config.Exec == nil {
+			continue
 		}
 
-		if msg.Config.EnvVars != nil {
-			static, dynamic = splitAppConfig(v.log, msg.Config.EnvVars)
+		// Get the config values, we use a iterator value of 0 here so that
+		// we will wait for at least one set of config values to resolve.
+		// In most virtual CEB cases we are only open for a single exec
+		// anyways so this resolves perfectly to our expected values.
+		env, _, err := w.Next(ctx, 0)
```

Comment:
So one thing I'm not sure of is if Next will return a value even if there are no config vars that were input. I could see a scenario where someone gets to this path without having set any config vars. I *think* the above `Update*` calls will prime it to generate an empty env to return for generation 1, but wanted to be sure.

Reply:
This is a really good point and no it won't. Let's gate on that.